### PR TITLE
Docusaurs updates on push

### PR DESCRIPTION
**Pull request by Surax98**
# Summary

I have updated the OpenAPI building action to update Docusaurus too, so any update in the API spec is reflected on the documentation. I had to use the same workflow because, using another resulted in overcomplicated steps to retrieve the JSON artifact file. Basically, now the openapi json file is generated, uploaded as an artifact and among release files and then docusaurus is built using this new openapi.json file instead of the old one, keeping it constantly updated if there is any change in the API spec.

TO DO: versioning documentation according to the application version

**Related issue :**

### CODE REVIEW

File: .github/workflows/build_openapi.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        path: ./docs/openapi/openapi.json
+
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    - name: Install dependencies
+      run: |
+        rm ./docs/openapi/openapi.json
+
+    - name: Download artifact from the latest workflow run
```
Comment (dciangot) [inline .github/workflows/build_openapi.yaml:59]: From here to the end of the file. Shouldn't this go into the gh page build action (pages.yaml)?
Or, should we remove pages.yaml?

**Review by dciangot — COMMENTED** [review]
@Surax98 please find one comment in code

Also, look like some commits are the same as #269 , is that right?

**Comment by Surax98** [comment]
@dciangot I based this PR on the #269 just because the OpenAPI generation script needs to be used for this PR. Just merge the first one and, if GitHub doesn't automatically remove already merged commits, I will reopen a new one

**Comment by dciangot** [comment]
No worries, I'll merge this directly thx

**Review by dciangot — APPROVED** [review]
LGTM

**Pull request merged**